### PR TITLE
Fix checking empty constant arrays in array function return type extensions

### PR DESCRIPTION
**Pull request by rvanvelzen**
Fix phpstan/phpstan#7928

### CODE REVIEW

File: src/Type/Php/ArrayKeyFirstDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -37,8 +37,11 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		if (count($constantArrays) > 0) {
 			$keyTypes = [];
 			foreach ($constantArrays as $constantArray) {
-				if ($constantArray->isEmpty()) {
+				$iterableAtLeastOnce = $constantArray->isIterableAtLeastOnce();
+				if (!$iterableAtLeastOnce->yes()) {
```
Comment (ondrejmirtes) [inline src/Type/Php/ArrayKeyFirstDynamicReturnTypeExtension.php:41]: The edits in these extensions should be tested too, right :)

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by herndlm** [comment]
Oh, did I break this? Thx!

**Comment by ondrejmirtes** [comment]
Yeah, I was a bit suspicious because `isEmpty()` cannot really be a `bool`, should be a `TrinaryLogic`, and the bug is caused by that :)

**Comment by ondrejmirtes** [comment]
Thank you very much!

**Pull request merged**